realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -154,11 +154,15 @@ public void addChangeListener(RealmChangeListener listener) {
      * Removes the specified change listener.
      *
      * @param listener the change listener to be removed.
+     * @throws IllegalStateException if you try to remove a listener from a non-Looper Thread.
      * @see io.realm.RealmChangeListener
      * @see #addChangeListener(RealmChangeListener)
      */
     public void removeChangeListener(RealmChangeListener listener) {
         checkIfValid();
+        if (handler == null) {
```

Comment:
Sure you can't add listeners, but you can still invoke `removeChangeListener` & `removeAllChangeListeners` on a non Looper thread. I think it's elegant to crash with an _IllegalStateException_ rather than a _NPE_